client/src/node/extension.ts

```diff
@@ -151,10 +151,19 @@ function triggerProfileUpdate(): void {
   const activeProfileName = profileConfig.getActiveProfile();
   if (activeProfileName in profileList || activeProfileName === "") {
     updateStatusBarProfile(activeProfileStatusBarIcon);
+
+    const connectionType =
+      profileList[activeProfileName].connectionType || ConnectionType.Rest;
+
+    //Set the connection type
+    commands.executeCommand("setContext", "SAS.connectionType", connectionType);
+
+    //See if the connection is direct (ie. serverId)
     commands.executeCommand(
       "setContext",
-      "SAS.connectionType",
-      profileList[activeProfileName].connectionType || ConnectionType.Rest
+      "SAS.connection.direct",
+      connectionType === ConnectionType.Rest &&
+        !!profileList[activeProfileName]?.serverId
```

Comment:
I was under the impression `serverId` existed on profile, or I'm missing it somewhere. It doesn't _seem_ that it does?